### PR TITLE
Fix Drones.remove

### LINKED ISSUE #23: Drones.remove(id) does not remove drones
This seems to possibly be something about how arrays and the delete keyword work in Javascript, but the result is that remove does not actually remove.

I only noticed this because I was fixing up the tests for the Drones class (see #21), and I couldn't work out why one of the expects on remove was failing, and it was because remove was actually faliing its test. Here is some example Javascript to actually explain what is happening.

```js
> const array = [{id: 0}, {id: 1}, {id: 2}]
// Returns undefined.
> array
// Returns [ { id: 0 }, { id: 1 }, { id: 2 } ]
> array.find(item => item.id = 1)
// Returns { id: 1}
> delete array.find(item => item.id = 1)
// Returns true
> array
// Returns [ { id: 0 }, { id: 1 }, { id: 2 } ]
```

So, as you can see delete doesn't remove things in arrays. One quick fix is simply to remove the object from the array, and for all I know that will remove the drone object from memory, but we should look into this, because if not a better solution should be found.

### PR DESCRIPTION
**Pull request by thewrongjames**
This commit fixes Drones.remove such that it actually removes drones. It uses
Array.splice rather than delete, as delete is actually only for removing object
properties, and doesn't work on items in an array.

This commit closes #23, and this issue and why this solution was used are
further discussed there.

### CODE REVIEW

**Comment by thewrongjames** [comment]
This commit will also fix the test failures that #25 is having, so if this pull request is acceptable, then it should be merged and the then drone-class-tests-fix should be rebased onto master, checked to make sure the tests do infact pass (which I have already done, by implementing this fix in the drone-class-tests-fix branch, though not pushing), and then #25 should also be merged to master, which will finally fix my screw up which landed us with broken tests on master, and that will finally resolve #21.

File: drones.js [hunk 1]
```diff
@@ -118,7 +118,7 @@ class Drones {
    */
   remove (id) {
     if (!this.containsId(id)) return false
-    delete this.getDrone(id)
+    this.all.splice(this.all.indexOf(this.getDrone(id)), 1)
```
Comment (ennuuos) [inline drones.js]: Please add a comment or change code style.

File: drones.js [hunk 2]
```diff
@@ -118,6 +118,18 @@ class Drones {
...
+     * internal droneList, which is accessed through this.all. It uses
+     * the Array.splice method, which takes a starting point and a
+     * distance, removes a sub-array which starts at the starting
+     * point and has a length of the distance from the array and then
+     * returns the sub array. For our purposes, the distance is only
+     * one, and the starting point is the index of the drone in
+     * this.all, which we get by calling indexOf on the array, and
+     * passing it the drone object which, inturn, we get from
+     * this.getDrone.
+     */
```
Comment (thewrongjames) [inline drones.js]: It actually took quite a bit to explain, which is a good sign that you were right about it requiring commenting or re-arranging. But, is this adequate?

Comment (ennuuos) [reply drones.js]: You don't think it's a bit much?
I would have been satisfied with "Gets the index of the drone with id and splices it out of the list"
But yes that is an acceptable explanatory comment even if it does take longer to read than it takes to decipher the code.
If you're happy I'm happy.

**Comment by thewrongjames** [comment]
Good to know, I will squash and merge now.

**Pull request merged**